### ISSUE TITLE
Is this extension usable?

### ISSUE THREAD

**Issue by DavidA2014** [montgomerybc/brief-keymap#1]
Hi, what is the status of this extension please? Is it usable?

**Comment by GaryWolfe** [comment]
I've only just given it a try, and it seems to be usable.  Download all contents in one folder: C:\Users\<USERID>\.vscode\extensions\brief-keymap-master

About the only thing I'd like to add to this is column-selection.

**Comment by AvidGameFan** [comment]
I've been playing with this, and it's partially usable.  A couple of problems I had were related to my environment and figuring out how to set up typescript, what folders go where, etc.  VS Code does not appear to run typescript for me, so I'm manually compiling in a CMD window after each modification.

Once the extension seemed to mostly work for me, some of the operations didn't seem to work right (for home and end); after a couple of minor code edits, they are both working fine for me.  (Could use some further tweaking, but very nice.)

If you don't care about the operations that require the typescript code, you can remove those mappings from package.json.  It's not like the triple-HOME press is an essential part of the keymapping, for example, but it's pretty slick if you're used to the old BRIEF keymap.

The main problem I'm having now is that Windows inserts special characters when I try to use ALT+PGUP, etc. , regardless of whether numlock is on.  I've spent a couple of hours Googling and this is apparently baked-in behavior.  Most search hits are for people who want to enable the feature.  I want to disable it!  (It would make sense to me if this worked only when numlock was on.)  So close, and yet so far.

**Comment by montgomerybc** [comment]
It's not terrible usable at all. I'm surprised you have found any value since I basically just did some of the easy mappings and stopped working on it.  (Can't believe it's been 2 years).
When I last did anything, the api's available for extensions weren't very helpful as far as positioning within a window (viewport?).  Those have since been improved, but I've been too lazy to pick it back up again.  Sorry

**Comment by AvidGameFan** [comment]
I'm grateful for the work you've done!  I fixed a couple of things, and while it's not "100% complete" it actually does most of what I'd like.  It's a fantastic start and brought a smile to my face to use it!  It's been a while since I've had an editor with the BRIEF commands, so I've been looking for a replacement.  For me, I think it's definitely of value.  (Is it weird to be excited about text editing?  ;-) )

If I am bored enough, I'll see if I can figure out what I did (and confirm that I actually improved it) and make a pull-request and maybe add to the documentation.  

While I'm sure more can always be done, you have to figure on how much you really need.  I mostly use the numeric keypad commands, and having +, -, Ins, Home, and End work are probably the biggest ones for me.  F5 for search, ALT-W for write....  I don't know if I need the command that split windows (F2?), but as the commands get more obscure, the less necessary they seem.  But my point is, you only have to get the most commonly-used mappings to work to make it worthwhile.

There's still a bug in the Home code, but I got it working well enough for now.

There's going to need to be some trickery to get the ALT-keypad commands to stop inserting special characters.

Gary

**Comment by montgomerybc** [comment]
Many of the things you mention are key bindings to some existing vscode functionality (those are the easy ones) i.e. F5 for search, ALT-W for save.  You wouldn't even need this extension for that.
For the ALT-Keypad stuff, do you mean dropping/jumping to bookmarks?  For that I cheat - there is an existing extension called Numbered Bookmarks (https://marketplace.visualstudio.com/items?itemName=alefragnani.numbered-bookmarks).  Then in my keybindings.json I put one of these for each numeric key:
`{ "key": "numpad0", "command": "numberedBookmarks.toggleBookmark0", "when": "editorTextFocus" },`
`{ "key": "ctrl+numpad0", "command": "numberedBookmarks.jumpToBookmark0" },`

**Comment by AvidGameFan** [comment]
Right, that built-in functionality makes most of it come together with the simple keymapping.  Pretty satisfied with the results.  

Regarding Alt-keypad, specifically, I was trying to use ALT-PGUP and ALT-PGDN.  I forgot about those commands (and tried them out when I saw them in your code), but that's kind of convenient for paging through code.  Unfortunately, if you hold the ALT key and type any of the numbers on the number pad, Windows 10 views that as attempting to enter a UNICODE character.  I don't recall what other BRIEF commands used ALT-keypad.  I don't think I used bookmarks, but thanks for the tip. I like how you can stack up multiple plugins in VS Code.

When I have more time, I'd like to see if column-selection can be added -- it looks like it's been added to VS Code already, so probably just needs another key remapping.

**Comment by zahroc** [comment]
I will have to give this a try. I am just getting back into programming again and have always used Brief as my editor or a Brief Keymapping. Started to look at VSC to see if I could use it.

**Comment by montgomerybc** [comment]
As long as your expectations are _very_ low.  I did a few of the easy
commands and have made an attempt at a couple of more useful ones.  It's
nowhere near production ready.  It _would_ be useful to know what your most
needed features are.  That way I can hopefully do them in the right order.

-Dave

**Comment by zahroc** [comment]
The features I use heavily are the following:

1) Column Selection (ALT-C)
2) Macro Recording (F7) and Playback (F8)
3) Cursor Movements (CTRL-<numerous>)
4) Delete Line (ALT-D)
5) Scrap Commands (Keypad Plus/Keypad Minus/Insert)
